astropy/units/deprecated.py

```diff
@@ -17,39 +17,25 @@
 
 """
 
-_ns = globals()
+from . import astrophys, cgs
+from .core import UnitBase, _add_prefixes, add_enabled_units, def_unit
+from .utils import generate_prefixonly_unit_summary, generate_unit_summary
 
+def_unit(["emu"], cgs.Bi, namespace=globals(), doc="Biot: CGS (EMU) unit of current")
+# Add only some *prefixes* as deprecated units.
+_add_prefixes(astrophys.jupiterMass, namespace=globals(), prefixes=True)
+_add_prefixes(astrophys.earthMass, namespace=globals(), prefixes=True)
+_add_prefixes(astrophys.jupiterRad, namespace=globals(), prefixes=True)
+_add_prefixes(astrophys.earthRad, namespace=globals(), prefixes=True)
 
-def _initialize_module():
-    # Local imports to avoid polluting top-level namespace
-    from . import astrophys, cgs
-    from .core import _add_prefixes, def_unit
-
-    def_unit(["emu"], cgs.Bi, namespace=_ns, doc="Biot: CGS (EMU) unit of current")
-
-    # Add only some *prefixes* as deprecated units.
-    _add_prefixes(astrophys.jupiterMass, namespace=_ns, prefixes=True)
-    _add_prefixes(astrophys.earthMass, namespace=_ns, prefixes=True)
-    _add_prefixes(astrophys.jupiterRad, namespace=_ns, prefixes=True)
-    _add_prefixes(astrophys.earthRad, namespace=_ns, prefixes=True)
-
-
-_initialize_module()
-
-
-###########################################################################
-# DOCSTRING
+__all__ = [name for (name, member) in globals().items() if isinstance(member, UnitBase)]
+__all__ += ["enable"]
```

Comment:
Tiny detail, but I noticed that in CDS, there is an `__all__` defined at the top, like 
```
__all__ = ["enable"]  #  Units are added at the end
```
which I think is a little clearer than adding it here, since people expect `__all__` to be at the top. (Indeed, in other modules an empty `__all__` is defined on top.)

Reply:
I thought it is better if both lines handling `__all__` are next to each other.

The real problem with this module is that it is named `deprecated`, but despite that it isn't. My understanding is that when this module was created there wasn't a good way for deprecating its contents because Python didn't have module `__getattr__()` yet, but it does now. If we deprecate this module then all the code in it will be temporary anyways. But this pull request will have to be merged first because I don't want to have to `import astropy.utils.decorators.deprecated as _deprecated`.

Reply:
Ah, nice larger plan!  I was already happy to have it merged as is, but now even more so. Let's just get it in!